src/index.cpp

```diff
@@ -1170,9 +1189,34 @@ void Index<T, TagT, LabelT>::search_for_point_and_prune(int location, uint32_t L
         for (auto &x : _pts_to_labels[location])
             filter_specific_start_nodes.emplace_back(_label_to_medoid_id[x]);
 
+        std::set<Neighbor> best_candidate_pool;
```

Comment:
should we use robin_set, and should we push this into scratch, to avoid creating local variables and alloc-ing memory each time we call insert point?

Reply:
Shall we create something like combined_candidates_pool, or just don't check if the pool is empty() and insert the candidates at the very back?